src/page-objects/administration/CustomerDetail.ts

```diff
@@ -1,24 +1,18 @@
 import type { Page, Locator } from '@playwright/test';
 import type { PageObject } from '../../types/PageObject';
-import type { components } from '@shopware/api-client/admin-api-types';
 
 export class CustomerDetail implements PageObject {
-    public readonly customerId;
-    public readonly customer: components['schemas']['Customer'];
-
     public readonly editButton: Locator;
     public readonly generalTab: Locator;
     public readonly accountCard: Locator;
 
-    constructor(public readonly page: Page, customer: components['schemas']['Customer'] ) {
-        this.customerId = customer.id;
-        this.customer = customer;
+    constructor(public readonly page: Page) {
         this.editButton = page.getByRole('button', { name: 'Edit' });
         this.generalTab = page.getByRole('link', { name: 'General' });
         this.accountCard = page.locator('.sw-customer-card')
     }
 
-    async goTo() {
-        await this.page.goto(`#/sw/customer/detail/${this.customerId}/base`);
+    url(customerId: string | undefined) {
```

Comment:
```suggestion
    url(customerId: string) {
```

Reply:
Did you use the insert type or the type that you get on `GET`? On insert the id is optional

Reply:
It comes from the definitions of the API client that we use. This should be the line which adds the types.

```
const customer = (await customerResp.json()) as { data: components['schemas']['Customer'] };
```

Reply:
Could we not define a new type that extends the type and makes it `string`?

Reply:
Sure, but then we have to do this kind of overrides more and more and even expose them. That could end up in a rabbit hole of fixing the actual entity schema on our end.

Reply:
@pweyck I added local overrides as discussed.